src/iter/repeat.rs

```diff
@@ -0,0 +1,109 @@
+use super::*;
+use super::internal::*;
+use std::usize;
+
+pub struct Repeat<T> {
+    element: T,
+}
+
+pub fn repeat<T: Clone>(elt: T) -> Repeat<T> {
+    Repeat { element: elt }
+}
+
+impl<T> ParallelIterator for Repeat<T>
+    where T: Clone + Send
+{
+    type Item = T;
+
+    fn drive_unindexed<C>(self, consumer: C) -> C::Result
+        where C: UnindexedConsumer<Self::Item>
+    {
+        bridge_unindexed(self, consumer)
+    }
+}
+
+impl<T> IndexedParallelIterator for Repeat<T>
+    where T: Clone + Send
+{
+    fn drive<C>(self, consumer: C) -> C::Result
+        where C: Consumer<Self::Item>
+    {
+        bridge(self, consumer)
+    }
+
+    fn with_producer<CB>(self, callback: CB) -> CB::Output
+        where CB: ProducerCallback<Self::Item>
+    {
+        callback.callback(RepeatProducer { repeat: self.element })
+    }
+
+    fn len(&mut self) -> usize {
+        usize::MAX
+    }
+}
+
+
+/// Producer
+
+struct RepeatProducer<P> {
```

Comment:
Nit: Can we make this `<T>` instead of `<P>`, so it matches the others? This confused me for a while, since I thought `P` must imply it is a producer or something. I'd also rather see `T: Clone + Send`.